### ISSUE TITLE
.NET8 upgrade from .NET Core 2.1 : Command cannot be used as type parameter 'TRequest' for 'IRequestHandler<TRequest>'

### ISSUE THREAD

**Issue by eb-datatel** [jbogard/MediatR#1033]
I cannot see what about this definition is not conforming to MediatR.IRequest.
```
The type 'EmployeeSignInCommand' cannot be used as type parameter 'TRequest' in the generic type or method 'IRequestHandler<TRequest>'.
There is no implicit reference conversion from 'EmployeeSignInCommand' to 'MediatR.IRequest'.
```
```
public class EmployeeSignInCommand : IRequest<SignInViewModel<EmployeeViewModel>>

public class EmployeeSignInCommandHandler : IRequestHandler<EmployeeSignInCommand, SignInViewModel<EmployeeViewModel>>
{
  public async Task<SignInViewModel<EmployeeViewModel>> Handle(EmployeeSignInCommand request, CancellationToken cancellationToken)
}
```

**Comment by eb-datatel** [comment]
It's no longer a compile error, but a runtime error. I also replaced my implementation of EmployeeSignInCommandHandler.Handle() with your minimalist version, and that also give the same error
```
An unhandled exception has occurred while executing the request.
      Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: The type 'DataTel.Application.Features.SignIn.Commands.EmployeeSignIn.EmployeeSignInCommand' cannot be used as type parameter 'TRequest' in the generic type or method 'MediatR.ISender.Send<TRequest>(TRequest, System.Threading.CancellationToken)'. There is no implicit reference conversion from 'DataTel.Application.Features.SignIn.Commands.EmployeeSignIn.EmployeeSignInCommand' to 'MediatR.IRequest'.
```

**Comment by eb-datatel** [comment]
It was not a MediatR issue. It was a change to .NET 8 where JSON serialization broke in two ways.
Newtonsoft.Json.JsonConvert.SerializeObject... started returning "{\"ValueKind\":1}" so I needed to switch to System.Text.Json

In .NET 8, **dynamic** is no longer supported.
It compiles, but at runtime, the object created this way looks fine in the debugger, but generates the type mismatch error I reported above.
`dynamic payload = JsonConvert.DeserializeObject(payloadJson, requestType); `
fortunately in this case it was only necessary to change **dynamic** to **var**
```
var payload = JsonConvert.DeserializeObject(payloadJson, requestType);
var result = Ok(await mediator.Send(payload));
```
